authentication/GetsWithRedirectThenPost.js

```diff
@@ -0,0 +1,157 @@
+// This script handles an authentication scheme with
+// - n GET requests giving HTTP 302 redirect while potentially collecting cookies
+// - then 1 POST request providing the login credentials to the server
+
+// Detailed usage tutorial and a php back-end can be found at https://github.com/ptrovatelli/ZAP-authentication-script-tutorial-GetsWithRedirectsThenPost
+
+// Make sure any Java classes used explicitly are imported
+var HttpRequestHeader = Java.type('org.parosproxy.paros.network.HttpRequestHeader');
+var HttpHeader = Java.type('org.parosproxy.paros.network.HttpHeader');
+var URI = Java.type('org.apache.commons.httpclient.URI');
+var AuthenticationHelper = Java.type('org.zaproxy.zap.authentication.AuthenticationHelper');
```

Comment:
Can't the `helper` parameters throughout be used?

Reply:
sorry I don't know what you mean here. can you give an example?